tools/tcptop.py

```diff
@@ -141,26 +154,56 @@ def range_check(string):
             &sk->__sk_common.skc_v6_rcv_saddr.in6_u.u6_addr32);
         bpf_probe_read_kernel(&ipv6_key.daddr, sizeof(ipv6_key.daddr),
             &sk->__sk_common.skc_v6_daddr.in6_u.u6_addr32);
-        ipv6_key.lport = sk->__sk_common.skc_num;
-        dport = sk->__sk_common.skc_dport;
+        bpf_probe_read_kernel(&ipv6_key.lport, sizeof(ipv6_key.lport),
+            &sk->__sk_common.skc_num);
+        bpf_probe_read_kernel(&dport, sizeof(dport),
+            &sk->__sk_common.skc_dport);
         ipv6_key.dport = ntohs(dport);
-        ipv6_send_bytes.increment(ipv6_key, size);
+        if (size >= 0)
+            ipv6_send_bytes.increment(ipv6_key, size);
     }
+    sock_store.delete(&tid);
     // else drop
 
     return 0;
 }
 
+int kretprobe__tcp_sendmsg(struct pt_regs *ctx)
+{
+    int size = PT_REGS_RC(ctx);
```

Comment:
Should we check `size <= 0` here? If that is the case, we just return 0? Similar to the original implementation, it is not needed to clear the sock_store map since next time when tid is reused, it will be updated first.

Reply:
> Should we check `size <= 0` here? If that is the case, we just return 0? Similar to the original implementation, it is not needed to clear the sock_store map since next time when tid is reused, it will be updated first.

thanks for your good suggestion. yes if the size <=0,  it need not to call the tcp_sendstat function. will update it   in next commit.